test/plugin_functional/test_suites/core_plugins/application_status.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import expect from '@kbn/expect';
+import {
+  AppNavLinkStatus,
+  AppStatus,
+  AppUpdatableFields,
+} from '../../../../src/core/public/application/types';
+import { PluginFunctionalProviderContext } from '../../services';
+import { CoreAppStatusPluginStart } from '../../plugins/core_app_status/public/plugin';
+import '../../plugins/core_provider_plugin/types';
+
+// eslint-disable-next-line import/no-default-export
+export default function({ getService, getPageObjects }: PluginFunctionalProviderContext) {
+  const PageObjects = getPageObjects(['common']);
+  const browser = getService('browser');
+  const appsMenu = getService('appsMenu');
+
+  const setAppStatus = async (s: Partial<AppUpdatableFields>) => {
+    await browser.executeAsync(async (status: Partial<AppUpdatableFields>, cb: Function) => {
+      const plugin = window.__coreProvider.start.plugins
+        .core_app_status as CoreAppStatusPluginStart;
+      plugin.setAppStatus(status);
+      cb();
+    }, s);
+  };
+
+  const navigateToApp = async (i: string) => {
+    return await browser.executeAsync(async (appId, cb: Function) => {
+      // navigating in legacy mode performs a page refresh
+      // and webdriver seems to re-execute the script after the reload
+      // as it considers it didn't end on the previous session.
+      // however when testing navigation to NP app, __coreProvider is not accessible
+      // so we need to check for existence.
+      if (!window.__coreProvider) {
+        cb();
+      }
+      const plugin = window.__coreProvider.start.plugins
+        .core_app_status as CoreAppStatusPluginStart;
+      try {
+        await plugin.navigateToApp(appId);
+        cb();
+      } catch (e) {
+        cb(e.message);
```

Comment:
optional: we can use an object to be more explicit `cb({error: e.message})`

Reply:
Done. As you already mentioned, our wrong signature on `browser.executeAsync` forces to cast when returning a typed thing fro the callback. But even the webdriver signature is quite wrong, as it does not explicitely specify the last `cb` parameter's type...
```typescript
// webdriver/index.d.ts
executeAsyncScript<T>(script: string|Function, ...var_args: any[]): Promise<T>;
```